designer/main.js

```diff
@@ -415,14 +415,18 @@ app.post('/intent', jsonParser, async function (req, res) {
 app.delete('/intent/:id', async function (req, res) {
     var id = req.params.id;
     
-    var serviceintent = db.data.serviceintents[id];
-    serviceintent.action = 'DELETE';
-    publishMetadata("ServiceIntent", serviceintent);       
-    
-    delete db.data.serviceintents[id];    
-    await db.write();    
-    
-    res.sendStatus(200)        
+    try {
+        var serviceintent = db.data.serviceintents[id];
+        serviceintent.action = 'DELETE';
+        publishMetadata("ServiceIntent", serviceintent);
```

Comment:
suggestion :- Better to use if check for Null Value.

Reply:
Yes, we could check the null value of the serviceintent variable. Please modify and test this.